homeassistant/components/pilight.py

```diff
@@ -0,0 +1,112 @@
+"""
+Component to create an interface to a Pilight daemon (https://pilight.org/).
+
+Pilight can be used to send and receive signals from a radio frequency
+module (RF receiver).
+
+RF commands received by the daemon are put on the HA event bus.
+RF commands can also be send with a pilight.send service call.
+"""
+# pylint: disable=import-error
+import logging
+import socket
+
+from homeassistant.helpers import validate_config
+from homeassistant.const import EVENT_HOMEASSISTANT_START
+from homeassistant.const import EVENT_HOMEASSISTANT_STOP
+from homeassistant.const import CONF_HOST, CONF_PORT
+
+REQUIREMENTS = ['pilight']
+
+DOMAIN = "pilight"
+EVENT = 'pilight_received'
+SERVICE_NAME = 'send'
+
+CONNECTED = False
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup(hass, config):
+    """Setup pilight component."""
+    from pilight import pilight
+
+    if not validate_config(config,
+                           {DOMAIN: [CONF_HOST, CONF_PORT]},
+                           _LOGGER):
+        return None
+
+    try:
+        pilight_client = pilight.Client(host=config[DOMAIN][CONF_HOST],
+                                        port=config[DOMAIN][CONF_PORT])
+    except (socket.error, socket.timeout) as err:
+        _LOGGER.error(
+            "Unable to connect to %s on port %s: %s",
+            config[CONF_HOST], config[CONF_PORT], err)
+        return None
+
+    # Start / stop pilight-daemon connection with HA start/stop
+    def start_pilight_client(_):
+        """Called once when home assistanz starts."""
+        pilight_client.start()
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_pilight_client)
+
+    def stop_pilight_client(_):
+        """Called once when home assistanz stops."""
+        pilight_client.stop()
+    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, stop_pilight_client)
+
+    def send_code(call):
+        """Send RF code to the pilight-daemon."""
+        message_data = call.data
+
+        if "protocol" not in message_data:
```

Comment:
You can remove this check since the service validation will take care that this code never gets called without a proper protocol.